### ISSUE TITLE
Support Annotations on Getters and Setters

### ISSUE THREAD

**Issue by gerhardj-b** [kbuntrock/openapi-maven-plugin#125]
Hi,

in our case, the models have Annotations - like `@NotNull` - only on the getters and setters, which are public.
Since some of the code is auto-generated, we also cannot change that behaviour easily.

Is it possible to take such Annotations into account - specifically when determining the required fields of a model?

**Comment by kbuntrock** [comment]
Hello.

Surely, it sounds do-able.

To ease my comprehension / the development : could you provide a sample class representative of what you have on your project?

**Comment by gerhardj-b** [comment]
Thanks for your quick response. I created a sample file that results in the said issue. I will paste it here down below. As I explained earlier, those files are auto-generated, namely with the help of `openapi-generator-maven-plugin`.

This file also shows another issue I found: A List field always leads to `uniqueItems` being set to true, although this is not neccessarily true in the source.

```java
@JsonPropertyOrder({
  SampleModel.JSON_PROPERTY_NAME,
  SampleModel.JSON_PROPERTY_DESCRIPTION,
  SampleModel.JSON_PROPERTY_TODOS
})
@jakarta.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen")
public class SampleModel {
  public static final String JSON_PROPERTY_NAME = "name";
  private String name;

  public static final String JSON_PROPERTY_DESCRIPTION = "description";
  private String description;

  public static final String JSON_PROPERTY_TODOS = "todos";
  private List<String> todos;

  public SampleModel() {
  }

  public SampleModel name(String name) {
    
    this.name = name;
    return this;
  }

   /**
   * Get name
   * @return name
  **/
  @jakarta.annotation.Nonnull
  @NotNull

  @JsonProperty(JSON_PROPERTY_NAME)
  @JsonInclude(value = JsonInclude.Include.ALWAYS)

  public String getName() {
    return name;
  }


  @JsonProperty(JSON_PROPERTY_NAME)
  @JsonInclude(value = JsonInclude.Include.ALWAYS)
  public void setName(String name) {
    this.name = name;
  }


  public SampleModel description(String description) {
    
    this.description = description;
    return this;
  }

   /**
   * Get description
   * @return description
  **/
  @jakarta.annotation.Nullable

  @JsonProperty(JSON_PROPERTY_DESCRIPTION)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public String getDescription() {
    return description;
  }


  @JsonProperty(JSON_PROPERTY_DESCRIPTION)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
  public void setDescription(String description) {
    this.description = description;
  }


  public SampleModel todos(List<String> todos) {
    
    this.todos = todos;
    return this;
  }

  public SampleModel addTodosItem(String todosItem) {
    if (this.todos == null) {
      this.todos = new ArrayList<>();
    }
    this.todos.add(todosItem);
    return this;
  }

   /**
   * Get todos
   * @return todos
  **/
  @jakarta.annotation.Nullable

  @JsonProperty(JSON_PROPERTY_TODOS)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public List<String> getTodos() {
    return todos;
  }


  @JsonProperty(JSON_PROPERTY_TODOS)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
  public void setTodos(List<String> todos) {
    this.todos = todos;
  }

```

**Comment by kbuntrock** [comment]
Thank you ! I'll be able to have a look on it next week.

**Comment by gerhardj** [comment]
Hi, I started an attempt to work on this issue. Could you probably give me some early feedback on it or tell me if this is going into the right direction? Then I will ofc give some unit tests a shot.

**Comment by kbuntrock** [comment]
Hello. Thank you @gerhardj for your contribution. :)
I will give feedbacks with pleasure but I'll be able to do so starting from the 02/05. I'm currently abroad without access to a computer to properly review code files.
Thank again for your interest and have a nice day.

Kind regards,
Kevin

**Comment by kbuntrock** [comment]
@gerhardj Thank you for the PR, I just read it. I need to think a bit about it, mainly the `getRelevantAnnotationsForField` in the Schema.class. I wonder if there is any way to do this part in a safer way.
Coming back to you soon.

**Comment by kbuntrock** [comment]
> This file also shows another issue I found: A List field always leads to `uniqueItems` being set to true, although this is not neccessarily true in the source.

Reacting also to this part : do you still have the issue with v0.0.19?

**Comment by gerhardj-b** [comment]
> > This file also shows another issue I found: A List field always leads to `uniqueItems` being set to true, although this is not neccessarily true in the source.
> 
> Reacting also to this part : do you still have the issue with v0.0.19?

no, `uniqueItems` behaves perfectly as expected now, many thanks for that!

**Comment by gerhardj** [comment]
Hi, I added a unit test for my changes. Feedback is very welcome, we would be very happy, if this functionality was available in the openapi-maven-plugin, since we really need to migrate from swagger-maven-plugin.

**Comment by kbuntrock** [comment]
Hello @gerhardj.

I merged your PR and released the v0.0.20 a couple of days ago. 😊

**Comment by gerhardj-b** [comment]
works perfectly!! many thanks @kbuntrock! 😊
